test/html/test_html.py

```diff
@@ -566,3 +566,31 @@ class PDF(FPDF, HTMLMixin):
         """
         )
         assert_pdf_equal(pdf, HERE / "html_description.pdf", tmp_path)
+
+
+def test_html_whitespace_handling(tmp_path):  # Issue 547
+    """Testing whitespace handling for write_html()."""
+    pdf = FPDF()
+    pdf.add_page()
+    pdf.write_html(
+        """
+<body>
+<h1>Issue 547 Test</h1>
+<p>
+<b>Testing   </b> paragraph blocks
+        that <i>span</i> <b>multiple lines</b>.
+    Testing tabs       and    spaces<br>
+    and break tags.<br>
+</p>
+<code>Testing code blocks with tabs      and    spaces.</code><br>
```

Comment:
`<pre><code>` blocks are also very common. Could you also add such block in this unit tests please?